frontend/spec/features/resources_spec.rb

```diff
@@ -0,0 +1,457 @@
+    fill_in 'resource_id_0_', with: "1 #{now}"
+    fill_in 'resource_id_1_', with: "2 #{now}"
+    fill_in 'resource_id_2_', with: "3 #{now}"
+    fill_in 'resource_id_3_', with: "4 #{now}"
+
+    element = find('#resource_lang_materials__0__language_and_script__language_')
+    element.click
+    element.send_keys('AU')
+    element.send_keys(:tab)
+
+    element = find('#resource_finding_aid_language_')
+    element.click
+    element.send_keys('ENG')
+    element.send_keys(:tab)
+
+    element = find('#resource_finding_aid_script_')
+    element.click
+    element.send_keys('Latn')
+    element.send_keys(:tab)
+
+    select 'Single', from: 'resource_dates__0__date_type_'
+    fill_in 'resource_dates__0__begin_', with: '1978'
+    select 'Collection', from: 'resource_level_'
+    fill_in 'resource_extents__0__number_', with: '10'
+    select 'Cassettes', from: 'resource_extents__0__extent_type_'
+
+    # Click on save
+    find('button', text: 'Save Resource', match: :first).click
+
+    expect(page).to have_text "Resource Resource Title #{now} created"
+
+    element = find('#tree-container')
+    expect(element).to have_text "Resource Title #{now}"
+  end
+
+  it 'reports warnings when updating a Resource with invalid data' do
+    now = Time.now.to_i
+    resource = create(:resource, title: "Resource Title #{now}")
+    run_index_round
+
+    visit "resources/#{resource.id}/edit"
+
+    fill_in 'resource_title_', with: ''
+
+    # Click on save
+    find('button', text: 'Save Resource', match: :first).click
+
+    within '#form_messages' do
+      element = find('.alert.alert-danger.with-hide-alert')
+      expect(element).to have_text 'Title - Property is required but was missing'
+    end
+  end
+
+  it 'reports errors if adding an empty child to a Resource' do
+    now = Time.now.to_i
+    resource = create(:resource, title: "Resource Title #{now}")
+    run_index_round
+
+    visit "resources/#{resource.id}/edit"
+
+    click_on 'Add Child'
+
+    # Click on save (+1)
+    find('#createPlusOne').click
+
+    within '#form_messages' do
+      element = find('.alert.alert-danger.with-hide-alert')
+      expect(element).to have_text 'Level of Description - Property is required but was missing'
+    end
+  end
+
+  it 'reports error if title is empty and no date is provided' do
+    now = Time.now.to_i
+    resource = create(:resource, title: "Resource Title #{now}")
+    run_index_round
+
+    visit "resources/#{resource.id}/edit"
+
+    click_on 'Add Child'
+
+    select 'Item', from: 'archival_object_level_'
+
+    # Click on save (+1)
+    find('#createPlusOne').click
+
+    within '#form_messages' do
+      element = find('.alert.alert-danger.with-hide-alert')
+      expect(element).to have_text 'Dates - one or more required (or enter a Title)'
+      expect(element).to have_text 'Title - must not be an empty string (or enter a Date)'
+    end
+  end
+
+  it 'can edit a Resource, add a second Extent, then remove it' do
```

Comment:
This test is missing the 'then remove it' part of the description. See [line 476](https://github.com/archivesspace/archivesspace/pull/3155/files?diff=unified&w=0#diff-0354bea0061358f490b907bf77fab55396b07159508b18cf33391cc4c4012bbcL476) of the original file to pick up where this leaves off.